src/reader/scanner.cpp

```diff
@@ -1,3 +1,124 @@
 #include "scanner.h"
+#include "token.h"
+#include <cctype>
+#include <iostream>
+#include <iterator>
+#include <memory>
+#include <ostream>
 
 Scanner::Scanner(std::string_view source) : source(source) {}
+
+std::unique_ptr<Identifier> Scanner::parse_identifier() {
+    int begin_index = current_index;
+
+    current_index++;
+
+    while (std::isalpha(source[current_index]) ||
+           std::isdigit(source[current_index])) {
+        current_index++;
+    }
+
+    int end_index = current_index;
+
+    std::string_view identifier =
+        source.substr(begin_index, (end_index - begin_index));
+
+    return std::make_unique<Identifier>(Identifier(identifier));
+}
+
+std::unique_ptr<Numeral> Scanner::parse_numeral() {
+    int begin_index = current_index;
+
+    current_index++;
```

Comment:
почему решил так не писать? менее читаемо типо?
```suggestion
    int begin_index = current_index++;
```